docs/source/examples/pt_zne.md

```diff
@@ -0,0 +1,378 @@
+
+from cirq import LineQubit, Circuit, CNOT, Ry, depolarize, X, Y, Z
+
+q0 = LineQubit(0)
+q1 = LineQubit(1)
+
+circuit = Circuit(CNOT(q0, q1))
+print(circuit)
+```
+```{code-cell} ipython3
+ptmcnot = ptm_matrix(circuit, 2)
+ax = sns.heatmap(ptmcnot.real, linewidth=0.5)
+print("Ideal CNOT PTM")
+plt.show()
+```
+```{code-cell} ipython3
+# PTM of a noisy CNOT gate: depolarizing noise
+noisy_circuit_incoherent = circuit.with_noise(depolarize(p=0.3))
+print(noisy_circuit_incoherent)
+
+ptmcnot = ptm_matrix(noisy_circuit_incoherent, 2)
+ax = sns.heatmap(ptmcnot.real, linewidth=0.5)
+print("\n Pauli Transfer Matrix of noisy CNOT (incoherent)")
+plt.show()
+```
+```{code-cell} ipython3
+# PTM of a noisy CNOT gate: Rz
+noisy_circuit_coherent = circuit.with_noise(Ry(rads=np.pi/12))
+print(noisy_circuit_coherent)
+
+ptmcnot = ptm_matrix(noisy_circuit_coherent, 2)
+ax = sns.heatmap(ptmcnot.real, linewidth=0.5)
+print("Pauli Transfer Matrix of noisy CNOT (coherent)")
+plt.show()
+```
+If we compare the PTM of the ideal CNOT gate to those when the gate was subjected to incoherent noise and coherent noise, 
+there are additional sources of errors to deal with when coherent noise is acting on the CNOT gate. These can be reduced or tailored to be close to how the incohrent noise PTM appears through Pauli Twirling.
+
+```{code-cell} ipython3
+from mitiq.pt import generate_pauli_twirl_variants
+
+# Generate twirled circuits
+NUM_TWIRLED_VARIANTS = 3
+twirled_circuits = generate_pauli_twirl_variants(
+    circuit, num_circuits=NUM_TWIRLED_VARIANTS)
+print("Example ideal twirled circuit", twirled_circuits[0], sep="\n")
+```
+Now, lets add coherent noise to the CNOT gate in each twirled circuit.
+```{code-cell} ipython3
+
+noisy_twirled_circuits = []
+
+for circ in twirled_circuits:
+    split_circuit = Circuit(circ[0], circ[1], Ry(rads=np.pi/12)(q0), Ry(rads=np.pi/12)(q1), circ[-1])
+    noisy_twirled_circuits.append(split_circuit)
+
+print("Example noisy twirled circuit", noisy_twirled_circuits[0], sep="\n")
+```
+
+The twirled PTM is averaged over each noisy twirled circuit such that the new PTM is close to that of the PTM of incoherent noise. We skip the step in this section as we require a very large number of twirled circuits to demonstrate the desired effect of averaging over multiple numpy arrays.
+
+## Noisy ZNE
+
+Lets define a larger circuit of CNOT, CZ and H gates. 
+
+```{code-cell} ipython3
+
+from cirq import LineQubit, Circuit, CZ, CNOT, H
+
+q0, q1, q2, q3  = LineQubit.range(4)
+circuit = Circuit(
+    H(q0),
+    CNOT.on(q0, q1),
+    CZ.on(q1, q2),
+    CNOT.on(q2, q3),
+)
+
+print(circuit)
+
+```
+
+We are going to add coherent noise to this circuit and then get the error-mitigated expectation value. For a detailed discussion on this, refer to the [ZNE user guide](../guide/zne-1-intro.md). 
+
+As we are using a simulator, we have to make sure the twirling gates are sandwiching the coherent noise channel in the circuit.
```

Comment:
This sentence is confusing, and in fact was already object of confusion with that infamous UnitaryHack task. The way to describe this is simply that we define a simulator where the noise model introduces coherent noise. Look at this as a black box, it being an actual hardware or a simulator. You ask the black box to execute the circuit, and whenever the box is supposed to execute a controlled gate, it corrupts the result with noise. If we mix it with twirling, or use words like sandwiching, etc.. we confuse the reader unnecessarily.